### PR TITLE
[CDAP-2360] fix internal errors in DatasetInstanceHandler for bad requests

### PR DESCRIPTION
**Pull request by anew**
- refactoring DatasetInstanceHandler to throw BadRequestException if the JSON is invalid, or if an ID is invalid, etc. 
- removed duplicate logging
- moved logging to after validation has happened
- added test case for all kinds of bad requests.

### CODE REVIEW

**Comment by anew** [comment]
Jira: https://issues.cask.co/browse/CDAP-2360
Build: http://builds.cask.co/browse/CDAP-RBT530-3

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java [hunk 1]
```diff
@@ -55,7 +56,7 @@
 /**
  * Handles dataset instance management calls.
  */
-public class DatasetInstanceService {
+public class DatasetInstanceService extends ConversionHelpers {
```
Comment (jwang47) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: Probably better to use static imports than to extend a helper class.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandlerTest.java [hunk 2]
```diff
@@ -345,7 +346,8 @@ private HttpResponse updateInstance(Id.DatasetInstance instance, Map<String, Str
 
   private ObjectResponse<List<DatasetSpecificationSummary>> getInstances(String namespace) throws IOException {
     return ObjectResponse.fromJsonBody(makeInstancesRequest(namespace),
-                                       new TypeToken<List<DatasetSpecificationSummary>>() { }.getType());
+                                       new TypeToken<List<DatasetSpecificationSummary>>() {
+                                       }.getType());
```
Comment (jwang47) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandlerTest.java]: Fix formatting

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java (hunk 1 above)
Comment (anew) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: Done.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandlerTest.java (hunk 2 above)
Comment (anew) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandlerTest.java]: Not even sure how that happened... must have been IDEA auto smarts. Fixed.

**Comment by anew** [comment]
@jwang47  I addressed your comments, please take another look. Thanks

**Comment by jwang47** [comment]
LGTM

**Comment by anew** [comment]
waiting for build to pass... it failed because of a flaky test :(

**Comment by anew** [comment]
Build succeeded. Merging.

**Pull request merged**